httpx/http_service.go

```diff
@@ -0,0 +1,165 @@
+package httpx
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/remind101/pkg/retry"
+
+	"golang.org/x/net/context"
+)
+
+type RoundTripper interface {
+	RoundTrip(*http.Request, context.Context) (*http.Response, error)
+}
+
+type Client struct {
+	innerClient		*http.Client
+	retryTransport 	*RetryTransport
```

Comment:
httpx.Client should be a pretty generic client, like http.Client. Maybe this?

```go
// Client is an extension of http.Client with context.Context support.
type Client struct {
        Transport RoundTripper
}

// NewClient returns a new Client instance that will use the given http.Client
// to perform round trips
func NewClient(c *http.Client) {
        return &Client{Transport: &Transport{Client: c}}
}

// Do performs the request and returns the response.
func (c *Client) Do(ctx context.Context, req *http.Request) (*http.Response, error) {
        return c.Transport.RoundTrip(ctx, req)
}

// Transport is an implementation of the RoundTripper interface that uses an
// http.Client from the standard lib.
type Transport struct {
        *http.Client
}
```

Now you can provide an implementation of the RoundTripper interface that retries:

```go
// RetryTransport is an implementation of the RoundTripper interface that
// retries requests.
type RetryTransport struct {
        *retry.Retrier
        Transport RoundTripper
}
```

And consumers can initialize an httpx.Client like this:

```go
c := &httpx.Client{
        Transport: &httpx.RetryTransport{
                Transport: &httpx.Transport{Client: http.Client},
        },
}
```